src/main/java/com/crowdin/cli/commands/actions/PreTranslateAction.java

```diff
@@ -141,6 +145,21 @@ private List<Long> prepareFileIds(Outputter out, PropertiesWithFiles pb, Crowdin
             .collect(Collectors.toList());
     }
 
+    private List<Long> prepareLabelIds(ProjectClient client) {
+        if (labelNames != null && !labelNames.isEmpty()) {
+            Map<String, Long> labels = client.listLabels().stream()
+                    .collect(Collectors.toMap(Label::getTitle, Label::getId));
+            labelNames.stream()
+                    .distinct()
+                    .forEach(labelName -> labels.computeIfAbsent(labelName, (title) -> client.addLabel(RequestBuilder.addLabel(title)).getId()));
+            return labelNames.stream()
+                    .map(labels::get)
+                    .collect(Collectors.toList());
+        } else {
+            return null;
+        }
+    }
```

Comment:
I just checked this feature in more detail and realized that this functionality should behave a bit differently than it does in the string add command. It shouldn't add a new label if one is missing in the Crowdin project. This command should give a warning if the non-existent label is passed.

Reply:
I've added this in the latest commit.